### ISSUE TITLE
Integration with morden IDE

### ISSUE THREAD

**Issue by pent0** [bentokun/symbian-clang#2]
- Visual Studio Code:
   - libtrk (pending)
   - gdb-arm-none-symbianelf (need binary + source code)

**Comment by fedor4ever** [comment]
what you need?
Symbian on device debugger named CODA use very specific protocol from Eclipse, can't remember  name. Also you can got Trk source and adapt. I saw them in symbian code in sources.

**Comment by fedor4ever** [comment]
CODA use tsf protocol - http://wiki.eclipse.org/TCF

**Comment by pent0** [comment]
I can use gdb, but in order to install SIS from computer, i need either TRK or CODA. Fortunately, runonphone from QT has the source code for sending messages to TRK and CODA, so i can port a part of them and make a library.

**Comment by fedor4ever** [comment]
@pent0 are you can debug from gdb? How?
What mean 

> **morden** IDE
never hear about one.

**Comment by pent0** [comment]
yeah, you cant, have to make a gdbstub wrapped on coda. sorry for confusion.
integration morden IDE likes VS or text editor like VSCode can be possible with plugin, it's just im not in the mood to do all of this (have my emulator also to work on)

if you can, i would love you to port runonphone, which contains all the Qt specific code, into seperate library for use :D

**Comment by fedor4ever** [comment]
> moRden IDE

you mean modeRn IDE? Haha.

Best way move to separate library trk library and write bindings then. I wish to deal with elf2e32 project and go further. Maybe build scummvm again =)

**Comment by pent0** [comment]
Oh no, I don't mean that you should drop on elf2e32!!!!!!!!!!! You can work
on the trk library at anytime, I'm not forcing you to do it :D

It would be one big step to debugging on real Symbian phone without crappy
IDE like Carbide or tie up to Qt. For debugging on computer, I'm working on
my emulator to run. It already have a gdbstub, it just needs an app to
display something.

Vào CN, 21 thg 10, 2018 vào lúc 14:45 Fiodar <notifications@github.com>
đã viết:

**Comment by pent0** [comment]
You are doing a great job on elf2e32 btw. Many crappy tools need a real port without crappy Windows bindings, the community will help, just take it easy :)

**Comment by fedor4ever** [comment]
@pent0, I will rewrite algorythm for generation symbian images once and all. I can rewrite this crap code endlessly but I have several project on mind =) I soon stop code refactoring not support itself. Anyway if you find some problems with elf2e32 feel free to bug report. Anyway if you wish you can refactor code for elf deals or describe on project wiki e32image structure =)

Why you dont like Carbide? It's awesome and handy. I wish to get Carbide based on modern Eclipse with dark theme ^^.
